at_secondary/at_persistence_secondary_server/lib/src/compaction/at_compaction_stats_impl.dart

```diff
@@ -0,0 +1,92 @@
+import 'dart:convert';
+
+import 'package:at_commons/at_commons.dart';
+import 'package:at_persistence_secondary_server/at_persistence_secondary_server.dart';
+import 'package:at_utils/at_logger.dart';
+
+class AtCompactionStatsImpl implements AtCompactionStats {
+  ///storing private constructor in private variable
+  ///private constructor refers to the instance of this class(only created once)
+  static final AtCompactionStatsImpl _singleton = AtCompactionStatsImpl._internal();
+
+  AtCompactionStatsImpl._internal();
+
+  ///returns the only object of [AtCompactionStatsImpl]
+  factory AtCompactionStatsImpl.init(atSign) {
+    _singleton.atSign = atSign;
+    _singleton.keyStore = SecondaryPersistenceStoreFactory.getInstance()
+        .getSecondaryPersistenceStore(atSign)
+        ?.getSecondaryKeyStore();
+    return _singleton;
+  }
+
+  factory AtCompactionStatsImpl.getInstance(atLogType) {
+   _singleton.atLogType = atLogType;
+    return _singleton;
+  }
+
+  late String atSign;
+  late AtLogType atLogType;
+  late DateTime compactionStartTime;
+  late Duration compactionDuration;
+  late int sizeBeforeCompaction;
+  late int sizeAfterCompaction;
+  late String compactionStatsKey;
+  late DateTime lastCompactionRun;
+  late int keysBeforeCompaction;
+  late int deletedKeysCount;
+  var keyStore;
+  final _logger = AtSignLogger("AtCompactionStats");
+
+  Future<void> _insertInitialStats() async {
+    try{
+      await keyStore?.put(compactionStatsKey, AtData()..data = json.encode({}));
+    } on Exception catch (_, e) {
+      _logger.severe(e);
+    }
+  }
+
+  ///measures pre-compaction parameters which will be further used for reference
+  @override
+  void preCompaction() {
```

Comment:
@kalluriramkumar @srieteja instead of void return type..should preCompaction return an internal object contain the pre compaction params.
e.g
PreCompaction {
int size;
DateTime startTime;
int keysBeforeCompaction;
String keyType;
}

Reply:
@murali-shris as this class is following a singleton pattern, would returning class object still make a difference?

Reply:
you need not return the singleton class object.
PreCompaction is a new class.